### PR TITLE
Add Python API for RMW implementation lookups

### PR DESCRIPTION
**Pull request by hidmic**
This pull request adds a basic Python API to lookup available RMW implementations in the system. Useful when parameterizing `pytests` by RMW implementation in pure Python packages i.e. no CMake templates available. 

Opening PR as a draft for preliminary discussion on whether this functionality should live here or in a separate package of its own.

### CODE REVIEW

File: rmw_implementation/rmw_implementation/__init__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+"""Utilities to discover available RMW implementations."""
+
+import ament_index_python
+
+
+def get_available_rmw_implementations():
+    """Return a list of all available RMW implementations as registered in the ament index."""
+    rmw_implementations = ament_index_python.get_resources('rmw_typesupport')
+    return [name for name, _ in rmw_implementations if name != 'rmw_implementation']
```
Comment (dirk-thomas) [inline rmw_implementation/rmw_implementation/__init__.py]: Have you tried to run this code?

Comment (hidmic) [reply rmw_implementation/rmw_implementation/__init__.py]: Hmm, that's funny. I did, though not recently. Removed that extra `, _` in https://github.com/ros2/rmw_implementation/pull/73/commits/7a0309c2aadf8b29308feaf46f3093811b9b92ca.

File: rmw_implementation/rmw_implementation/__init__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+"""Utilities to discover available RMW implementations."""
+
+import ament_index_python
+
+
+def get_available_rmw_implementations():
+    """Return a list of all available RMW implementations as registered in the ament index."""
+    rmw_implementations = ament_index_python.get_resources('rmw_typesupport')
+    return [name for name in rmw_implementations if name != 'rmw_implementation']
```
Comment (dirk-thomas) [inline rmw_implementation/rmw_implementation/__init__.py]: How about returning a set here to not imply any order?

Comment (hidmic) [reply rmw_implementation/rmw_implementation/__init__.py]: Sounds perfectly reasonable, see https://github.com/ros2/rmw_implementation/pull/73/commits/5a7690aebcd12e6f920a65b474076eac48ef1350.

**Comment by hidmic** [comment]
Getting this one in to re-run CI on https://github.com/ros2/ros2cli/pull/304.

**Comment by wjwwood** [comment]
We should consider moving this to a different package. This change causes all of the client libraries to depend on Python indirectly, which probably isn't ideal... If it were a separate package, them packages like rcl or rclcpp could still depend on and use this python code, but they could do so via a build or buildtool depend (I'm pretty sure it's not being used by them at runtime). But since this is an exec depend and `rmw_implementation` is most often an exec depend (perhaps transitively) it causes a runtime dependency on `ament_index_python` and therefore Python itself.

**Comment by dirk-thomas** [comment]
@wjwwood Can you please create a separate ticket for this. Otherwise the chance is high that your comment on a closed ticket will get lost.

**Pull request merged**